src/NUnitFramework/tests/Attributes/ValueSourceTests.cs

```diff
@@ -168,6 +172,52 @@ public static IEnumerable<int> IntegerProvider()
 
                 return dataList;
             }
+
+            public static IEnumerable<int> ForeignNullResultProvider()
+            {
+                return null;
+            }
+        }
+
+        public static string NullSource = null;
+
+        public static IEnumerable<int> NullDataSourceProvider()
+        {
+            return null;
+        }
+
+        public static IEnumerable<int> NullDataSourceProperty
+        {
+            get { return null; }
+        }
+
+        [Test, Explicit("Null or nonexisting data sources definitions should not prevent other tests from run #1121")]
+        public void ValueSourceMayNotBeNull(
+            [ValueSource("NullSource")] string nullSource,
+            [ValueSource("NullDataSourceProvider")] string nullDataSourceProvided,
+            [ValueSource(typeof(ValueProvider), "ForeignNullResultProvider")] string nullDataSourceProvider,
+            [ValueSource("NullDataSourceProperty")] int nullDataSourceProperty,
+            [ValueSource("SomeNonExistingMemberSource")] int nonExistingMember)
+        {
+            Assert.Fail();
+        }
+
+        [Test]
+        public void ValueSourceAttributeShouldThrowInsteadOfReturningNull()
+        {
+            var method = new MethodWrapper(GetType(), "ValueSourceMayNotBeNull");
+            var parameters = method.GetParameters();
+
+            foreach (var parameter in parameters)
+            {
+                var dataSource = parameter.GetCustomAttributes<IParameterDataSource>(false)[0];
+
+                Assert.Throws<InvalidDataSourceException>(() =>
+                {
+                    var data = dataSource.GetData(parameter);
+                    Assert.IsNotNull(data);
```

Comment:
I am assuming that the GetData is the line that throws, so is the Assert.IsNotNull necessary?